js-peer/src/components/message.tsx

```diff
@@ -1,16 +1,70 @@
+import React, { useEffect } from 'react'
 import { useLibp2pContext } from '@/context/ctx'
-import React, { useCallback, useEffect, useRef, useState } from 'react'
-import { ChatMessage } from '@/context/chat-ctx'
-import Blockies from 'react-18-blockies'
+import { ChatMessage, useChatContext } from '@/context/chat-ctx'
+import { Peer } from './peer'
+import { peerIdFromString } from '@libp2p/peer-id'
 
-interface MessageProps extends ChatMessage {}
+interface Props extends ChatMessage {
+  dm: boolean
+}
+
+export const Message = ({
+  msgId,
+  msg,
+  fileObjectUrl,
+  from,
+  peerId,
+  read,
+  dm,
+  receivedAt,
+}: Props) => {
+  const {
+    messageHistory,
+    setMessageHistory,
+    directMessages,
+    setDirectMessages,
+  } = useChatContext()
 
-export function MessageComponent({ msg, fileObjectUrl, from, peerId }: MessageProps) {
   const { libp2p } = useLibp2pContext()
 
+  useEffect(() => {
+    if (read) {
+      return
+    }
+
+    const updateMessages = (messages: ChatMessage[]) =>
```

Comment:
I'd move this out of the component and add the `msgId` as an argument. 
```suggestion
    const markAsRead = (messages: ChatMessage[]) =>
```

Reply:
moved this to a hook as it needs to run when the message is rendered